src/ui/public/query_manager/index.js

```diff
@@ -0,0 +1,79 @@
+import _ from 'lodash';
+import { FilterManagerProvider } from 'ui/filter_manager';
+import { FilterBarLibMapAndFlattenFiltersProvider } from 'ui/filter_bar/lib/map_and_flatten_filters';
+import { FilterBarLibExtractTimeFilterProvider } from 'ui/filter_bar/lib/extract_time_filter';
+import { FilterBarLibChangeTimeFilterProvider } from 'ui/filter_bar/lib/change_time_filter';
+import { toKueryExpression, fromKueryExpression, nodeTypes, filterToKueryAST } from 'ui/kuery';
+
+export function QueryManagerProvider(Private) {
+  const filterManager = Private(FilterManagerProvider);
+  const mapAndFlattenFilters = Private(FilterBarLibMapAndFlattenFiltersProvider);
+  const extractTimeFilter = Private(FilterBarLibExtractTimeFilterProvider);
+  const changeTimeFilter = Private(FilterBarLibChangeTimeFilterProvider);
+
+  return function (state) {
+
+    function add(field, values = [], operation, index) {
```

Comment:
The idea is that this will become the API we pass to clients who want to manipulate the query/filters. In particular, the vis team needs an API to pass to visualizations. Mostly I was being lazy here because I'm going to have to revisit and expand this API before handing it to the vis team. I figured I'd get a better idea of what the function signatures should look like once I start working on the rest of the API.